src/decimal128.ts

```diff
@@ -331,7 +335,7 @@ export class Decimal128 extends BSONValue {
     // to represent user input
 
     // Overflow prevention
-    if (exponent <= radixPosition && radixPosition - exponent > 1 << 14) {
+    if (exponent <= radixPosition && radixPosition > exponent + (1 << 14)) {
```

Comment:
@W-A-James Double check that these changes are necessary

Reply:
@nbbeeken Taking another look, this was just reorganizing the inequality for readibility's sake, so I'd prefer to keep this change